content/module-3/1-create-your-messaging-protocol.md

```diff
@@ -0,0 +1,251 @@
+
+<HighlightBox type="info" title="Example: ERC20Messaging">
+
+**ERC20Messaging** exposes the `sendToken` function, as we have seen before, which:
+
+1- Burns the requested amount of the requested token
+
+2- Emits the **TokenSent** event, i.e., the arbitrary event that carries the messaging protocol specific data used on the receiving subnet to execute the cross-subnet message
+
+3- Emits the **CrossSubnetMessageSent** core event to announce the cross-subnet message
+
+```
+function sendToken(SubnetId targetSubnetId, string calldata symbol, address receiver, uint256 amount) external {
+    if (_toposCoreAddr.code.length == uint256(0)) revert InvalidToposCore();
+    Token memory token = getTokenBySymbol(symbol);
+    _burnTokenFrom(msg.sender, symbol, amount);                           // 1- Burn the requested amount of the requested token
+    emit TokenSent(targetSubnetId, symbol, token.addr, receiver, amount); // 2- Emit the TokenSent event
+    _emitMessageSentEvent(targetSubnetId);                                // 3- Emit the CrossSubnetMessageSent event
+}
+```
+
+</HighlightBox>
+
+Only the last line that emits the **CrossSubnetMessageEvent** is a requirement for all messaging protocols. So if we were to name the first function of **CustomMessaging** `doSomething`, `doSomething` would have the following structure:
+
+```
+function doSomething(SubnetId targetSubnetId, ...otherArgs) external {
+    ...                                    // 1- Do something
+    ...                                    // 2- Emit an event that describes the something
+    _emitMessageSentEvent(targetSubnetId); // 3- Emit the CrossSubnetMessageSent event
+}
+```
+
+### Receiving subnet (2)
+
+Let's now take a look at the second function (`_execute`) signature:
+
+```
+function _execute(
+    uint256[] memory logIndexes,
+    address[] memory logsAddress,
+    bytes[] memory logsData,
+    bytes32[][] memory logsTopics,
+    SubnetId networkSubnetId
+) internal virtual {}
+```
+
+As detailed before, this function's signature is defined in **ToposMessaging**'s interface and the function is itself called by **ToposMessaging**'s `execute` function, so as a messaging protocol developer, you need not care about how all these arguments are retrieved. You only need to care about how you use them.
+
+You see that, except `networkSubnetId`, all arguments are data projected against a certain log index.
+
+With `logIndexes`, you can retrieve the index of the logs that semantically will describe the cross-subnet message and how it should be executed on the receiving subnet.
+
+<HighlightBox type="tip" title="Which log indexes should I use?">
+
+Remember that because you use the same messaging protocol contract on the sending and receiving subnets, and you developped that messaging protocol, you know which log leads to what information. 
```

Comment:
```suggestion
Remember that because you use the same messaging protocol contract on the sending and receiving subnets, and you developed that messaging protocol, you know which log leads to what information. 
```